lua/harpoon/ui.lua

```diff
@@ -126,6 +135,13 @@ function HarpoonUI:toggle_quick_menu(list)
 
     local contents = self.active_list:display()
     vim.api.nvim_buf_set_lines(self.bufnr, 0, -1, false, contents)
+    if self.settings.select_current then
+        for idx, line in ipairs(contents) do
+            if line == currentFileName then
+                vim.api.nvim_win_set_cursor(self.win_id, { idx, 0 })
+            end
```

Comment:
You are setting the cursor, but not applying highlights. I would suggest to move the if statement inside the loop:

```lua
for idx, line in ipairs(contents) do
  if line == currentFileName  then
    vim.api.nvim_buf_add_highlight(...)
    if self.settings.select_current then
      vim.api.nvim_win_set_cursor(...)
    end
    break -- also make sure to break after you find the line
  end
end
```

Reply:
Done